Sources/Workspace/ManagedDependency.swift

```diff
@@ -14,195 +14,188 @@ import SourceControl
 import TSCBasic
 import TSCUtility
 
-/// An individual managed dependency.
-///
-/// Each dependency will have a checkout containing the sources at a
-/// particular revision, and may have an associated version.
-public class ManagedDependency {
-    /// Represents the state of the managed dependency.
-    public enum State: Equatable {
-
-        /// The dependency is a managed checkout.
-        case checkout(CheckoutState)
-
-        /// The dependency is in edited state.
-        ///
-        /// If the path is non-nil, the dependency is managed by a user and is
-        /// located at the path. In other words, this dependency is being used
-        /// for top of the tree style development.
-        case edited(AbsolutePath?)
+extension Workspace {
+    /// An individual managed dependency.
+    ///
+    /// Each dependency will have a checkout containing the sources at a
+    /// particular revision, and may have an associated version.
+    final public class ManagedDependency {
```

Comment:
FWIW — In my parallel work on #3640, I changed `ManagedDependency` to a structure, for which Swift could automatically synthesize `Equatable` conformance. To make the recursive `edited` case work, I made `State` an `indirect enum`.

Reply:
@mattt how about we carve out a separate PR just to deal with improving ManagedDependency (struct, bashedOn as associated value, etc). then rebase both of our PRs on top?